### PR TITLE
feat: allow single checkbox and radio buttons as field options in conf

### PR DESCRIPTION
**Pull request by wizehive-chrismanjoine**
This introduces 2 new field configs one for single checkboxes and one for radio selects. I did not do multiple checkbox implementations as I do not currently see a use case for them. 

- **checkboxes** #6 now has a _label_ attribute that can be passed which will be included when highlighted, so that it doesn't display the full question. (I could have reversed the name, help implementation but wanted to avoid confusion with that. I would be willing to name it something like _hightlightLabel_ if that is more readable)

- **radio** #5 introduces an option for _inline_ vs standard so that you can either display the checkboxes in one row on the form or as separate rows for each radio selection. This also uses the _label_ for highlights like checkboxes to allow having a more summarizable name for the highlights.

Added the ability to  _disable_ a dropdown item, this allows us to put a divider in the dropdown to breakup entries. This is something that is part of the newer angular 1.x ng-options, but since our version is older I had to implement the ng-repeat pattern on the dropdown template. 

There is also a bug in the code that does not remove highlights when they are removed. I unset those before they are saved again to the system in plugin.controller. line 789-796

### CODE REVIEW

File: src/inputs/dropdown.html [hunk 1]
```diff
@@ -2,7 +2,8 @@
 <script type="text/ng-template" id="wgn-config-input-dropdown">
 	<select class="form-control" name="{{ field.id }}"
 			ng-model="editing.config[field.id]"
-			ng-required="field.required"
-			ng-options="opt.value as opt.label for opt in field.options">
+			ng-required="field.required">
+			<option ng-repeat="item in field.options" ng-disabled="item.disable" ng-selected="{{item.value == editing.config[field.id]}}" value="{{item.value}}">{{item.label}}</option>
 	</select>
+
```
Comment (tehpsalmist) [inline src/inputs/dropdown.html]: remove newline

File: src/inputs/radio.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+			<input type="radio" value="{{item.value}}" ng-model="editing.config[field.id]">
+			{{item.label}}
+		</label>
+	</div>
+
+	<label class="radio-inline" ng-repeat="item in field.options"  ng-if="field.inline">
+		<input type="radio" value="{{item.value}}" ng-model="editing.config[field.id]">
+		{{item.label}}
+	</label>
+
```
Comment (tehpsalmist) [inline src/inputs/radio.html]: remove newline

File: src/main.controller.js [hunk 3]
```diff
@@ -708,7 +708,26 @@ plugin.controller('wgnConfigCtrl', ['$scope', '$q', '$routeParams', 'znData', 'z
 							});
 						}
 						break;
-
+					case 'checkbox':
+						var checkDef = $scope.options.getField(input.id);
+						if ($scope.editing.config[input.id]) {
```
Comment (tehpsalmist) [inline src/main.controller.js]: wrap this if block in newlines

File: src/main.controller.js [hunk 4]
```diff
@@ -708,7 +708,26 @@ plugin.controller('wgnConfigCtrl', ['$scope', '$q', '$routeParams', 'znData', 'z
...
+							formatedHighligts.push({
+								type: checkDef.label,
+								value: "Yes"
+							});
+						}
+						break;
+					case 'radio':
+						var radioDef = $scope.options.getField(input.id);
+						if ($scope.editing.config[input.id]) {
```
Comment (tehpsalmist) [inline src/main.controller.js]: wrap if block in newlines

File: src/plugin.scss [hunk 5]
```diff
@@ -10,6 +10,7 @@ $color-success: #3c763d;
 		border-color: $color;
 		box-shadow: inset 0 1px 1px rgba(0, 0, 0, .075);
 	}
+
```
Comment (tehpsalmist) [inline src/plugin.scss]: probably unnecessary, but I hate SASS, so ignore me.

File: src/plugin.scss [hunk 6]
```diff
@@ -137,7 +138,23 @@ $color-success: #3c763d;
...
+			font-size: .9em;
+			font-weight: 100;
 		}
+
+		.form-check-label {
+			line-height: initial;
+			font-size: .9em;
+			font-weight: 100;
+		}
+
```
Comment (tehpsalmist) [inline src/plugin.scss]: same as last comment. I don't prefer this newline, but whatever.

File: src/main.controller.js (hunk 4 above)
Comment (wizehive-chrismanjoine) [reply src/main.controller.js]: done

File: src/plugin.scss (hunk 5 above)
Comment (wizehive-chrismanjoine) [reply src/plugin.scss]: ha I love SASS personally

File: src/main.controller.js (hunk 3 above)
Comment (wizehive-chrismanjoine) [reply src/main.controller.js]: done

**Comment by tehpsalmist** [comment]
I'll let Anna sign off on the whole deal, in case she has anything interesting to say.

File: src/inputs/checkbox.html [hunk 7]
```diff
@@ -0,0 +1,5 @@
+<!--Checkbox input-->
+<script type="text/ng-template" id="wgn-config-input-checkbox">
+	<input type="checkbox"  name="{{ field.id }}" ng-model="editing.config[field.id]">
+	<label class="form-check-label">{{field.name}}</label>
```
Comment (AParks) [inline src/inputs/checkbox.html]: @wizehive-chrismanjoine minor nitpick, can we put the input inside the label, so you when you click on the text, it selects/deselects the checkbox?

Comment (tehpsalmist) [reply src/inputs/checkbox.html]: The behavior will also be achieved by using `for` on the label and referring it to the input's `id`, assuming it is unique. But I'm with Anna, in that the current input patterns should probably be preferred unless we spot a bug in those patterns that needs adjustment.

Comment (wizehive-chrismanjoine) [reply src/inputs/checkbox.html]: This has been updated. not sure why I did it that way to begin with.

**Pull request merged**